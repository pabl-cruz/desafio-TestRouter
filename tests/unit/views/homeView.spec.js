import { shallowMount } from '@vue/test-utils'
import { createRouter, createWebHistory} from 'vue-router'
import HomeView from '@/views/HomeView.vue'

describe('HomeView', () => {
    const wrapper = shallowMount(HomeView)
    it('captura de contenido HTML en prueba y contrasta', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})

test('Comprobando ruta a la vista Homeview', async () => {
    const routerTest = createRouter({
        history: createWebHistory(),
        routes: [{
            path: '/home',
            name: 'home',
            component: HomeView
        }
        ]
    })
    routerTest.push('./home')
    await routerTest.isReady()
    const wrapper = shallowMount(HomeView, {
        global: {
            plugins: [routerTest]
        }
    })
    expect(wrapper.findComponent(HomeView).exists()).toBeTruthy()
})
