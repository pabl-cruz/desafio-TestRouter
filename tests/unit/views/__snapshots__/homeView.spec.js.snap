// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView captura de contenido HTML en prueba y contrasta 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <posts-view-stub></posts-view-stub>
</div>
`;
